visualizer/main.py

```diff
@@ -0,0 +1,399 @@
+import pickle
+import os.path
+
+from googleapiclient import discovery
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ['https://www.googleapis.com/auth/drive']
+
+class DataPoint:
+    pass
+
+class Parser:
+
+    def __init__(self, file_path, begin, end):
+        simulations = []
+        config_data = []
+        with open(file_path) as f:
+            for line in f:
+                # found start of section so start iterating from next line
+                if line.startswith(begin):
+                    simulation = []
+
+                    for line in f:
+                        # found end so end function
+                        if line.startswith(end):
+                            simulations.append(simulation)
+                            break
+
+                        elif line.startswith('*'):
+                            config_data.append(line)
+
+                        dp = self.string_to_datapoint(line)
+                        # yield every line in the section
+                        simulation.append(dp)
+
+        self.config = config_data
+        self.sims = simulations
+
+    def string_to_datapoint(self, line) -> DataPoint:
+
+        tokens = line.split(',')
+        type = tokens[0]
+        dp_params = tokens[1:]
+
+        if type == 'a':
+            return ArbiterDataPoint(arbiter_id=dp_params[0], consumer_id=dp_params[1], cycle_number=dp_params[2])
+        elif type == 'c':
+            return ConsumerDataPoint(consumer_id=dp_params[0], operation=dp_params[1], cycle_number=dp_params[2])
+
+class SheetRenderer:
+
+    def __init__(self, service, sims, config):
+        self.l_c_dict = {
```

Comment:
These dictionaries should be static. Move them out of init and rename them all caps.